src/nxt_http_return.c

```diff
@@ -38,30 +50,16 @@ nxt_http_return_init(nxt_mp_t *mp, nxt_http_action_t *action,
 
     conf->status = nxt_conf_get_number(acf->ret);
 
-    if (acf->location.length > 0) {
-        if (nxt_is_complex_uri_encoded(acf->location.start,
-                                       acf->location.length))
-        {
-            loc = nxt_str_dup(mp, &conf->location, &acf->location);
-            if (nxt_slow_path(loc == NULL)) {
-                return NXT_ERROR;
-            }
-
-        } else {
-            loc = &conf->location;
-
-            encode = nxt_encode_complex_uri(NULL, acf->location.start,
-                                            acf->location.length);
-            loc->length = acf->location.length + encode * 2;
-
-            loc->start = nxt_mp_nget(mp, loc->length);
-            if (nxt_slow_path(loc->start == NULL)) {
-                return NXT_ERROR;
-            }
-
-            nxt_encode_complex_uri(loc->start, acf->location.start,
-                                   acf->location.length);
-        }
+    conf->location = nxt_var_compile(&acf->location, mp, 0);
```

Comment:
It's better to keep previous main logic.

```
if (acf->location.length > 0) {
     conf->location = nxt_var_compile(&acf->location, mp, 0);
}
```
Then, the condition `conf->location != NULL` can be used on runtime phase.

Reply:
I changed it considerably.  It still uses that variable, however.  Otherwise I needed to add more code.

We might consider using `nxt_var_is_const()` each time if you want to remove the variable from the structure, but I think letting `conf->location` be `NULL` adds more code than it removes.

Reply:
If users don't set "location", it shouldn't call anything in "location", like `nxt_http_return_encode()`. But it seems it's called when "location" is not set.

Reply:
Oops, that's right.

Reply:
I'm a bit confused about the previous code:

<https://github.com/nginx/unit/blob/aeed86c6829c62359e79f239b849766efb8857a7/src/nxt_http_return.c#L41>

Isn't `acf->location` supposed to be `NULL` if the user doesn't set `"location"`?